internal/ice/util.go

```diff
@@ -18,21 +18,36 @@ func localInterfaces() (ips []net.IP) {
 		if iface.Flags&net.FlagLoopback != 0 {
 			continue // loopback interface
 		}
+
 		addrs, err := iface.Addrs()
 		if err != nil {
 			return ips
 		}
+
 		for _, addr := range addrs {
 			var ip net.IP
-			switch v := addr.(type) {
+			switch addr := addr.(type) {
 			case *net.IPNet:
-				ip = v.IP
+				ip = addr.IP
 			case *net.IPAddr:
-				ip = v.IP
+				ip = addr.IP
+
 			}
+
 			if ip == nil || ip.IsLoopback() {
 				continue
 			}
+
+			// The conditions of invalidation written below are defined in
+			// https://tools.ietf.org/html/rfc8445#section-5.1.1.1
+			if _ip := ip.To4(); _ip != nil {
```

Comment:
I would do this, so we can keep this function in the same style. I don't have an opinion either way, but just so we don't introduce bugs.

Also can you squash the three commits, just so if anything happens it is easier to debug in logical chunks. Thank you!

```
if ipv4 := ip.To4(); ipv4 == nil {
  if !isSupportedIPv6(ip) {
￼      continue
￼  }
}

   ips = append(ips, _ip)
```